Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -859,19 +859,28 @@ void MediaPlayerPrivateGStreamerMSE::markEndOfStream(MediaSourcePrivate::EndOfSt
     updateStates();
 }
 
+void MediaPlayerPrivateGStreamerMSE::unmarkEndOfStream()
+{
+    GST_DEBUG("Unmarking end of stream");
+    m_eosPending = false;
+}
+
 MediaTime MediaPlayerPrivateGStreamerMSE::currentMediaTime() const
 {
+    float cachedPosition = m_cachedPosition;
     MediaTime position = MediaPlayerPrivateGStreamer::currentMediaTime();
+    float playbackProgress = std::abs(position.toFloat() - cachedPosition);
 
-    if (m_eosPending && (paused() || (position >= durationMediaTime()))) {
```

Comment:
In case when video is pushed for 1s but audio much more, like 60s, After endOfStream durationMediaTime() will be highest value, 60s.
But position is fetched from video decoder and it remains the same, e.g. is stuck due to video playback is already finished. But audio is still playing.

So after changing duration before MSE.endOfStream() to lower value than current, we should expect to remove all buffers more than new duration, it will produce durationMediaTime() to return proper value.